test_package/conanfile.py

```diff
@@ -11,6 +11,7 @@ class TestPackageConan(ConanFile):
 
     def build(self):
         cmake = CMake(self)
+        cmake.definitions['WITH_GMOCK'] = self.options['gtest'].build_gmock
```

Comment:
If we declare this from the main package, we won't need to do this.

Reply:
(my comment above applies here too)